.github/workflows/cloud_integration.yml

```diff
@@ -8,72 +8,62 @@ on:
     - main
 env:
   GH_ANNOTATION: true
+  DOCKER_BUILDKIT: 1
 jobs:
-  # todo: Keep in sync with `release.yml`
   docker_build:
-    name: Docker build
     runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        target: [proxy, controller, web, cni-plugin, debug, cli-bin, grafana]
+    name: Docker build (${{ matrix.target }})
     steps:
     - name: Checkout code
       # actions/checkout@v2
       uses: actions/checkout@722adc6
-    - name: Setup SSH config for Packet
+    - name: Set environment variables from scripts
       run: |
-        mkdir -p ~/.ssh/
-        touch ~/.ssh/id && chmod 600 ~/.ssh/id
-        echo "${{ secrets.DOCKER_SSH_CONFIG }}"  > ~/.ssh/config
-        echo "${{ secrets.DOCKER_PRIVATE_KEY }}" > ~/.ssh/id
-        echo "${{ secrets.DOCKER_KNOWN_HOSTS }}" > ~/.ssh/known_hosts
-        ssh linkerd-docker docker version
+        . bin/_tag.sh
+        echo ::set-env name=TAG::$(CI_FORCE_CLEAN=1 bin/root-tag)
+
+        . bin/_docker.sh
+        echo ::set-env name=DOCKER_REGISTRY::$DOCKER_REGISTRY
+        echo ::set-env name=DOCKER_BUILDKIT_CACHE::${{ runner.temp }}/.buildx-cache
+    - name: Cache docker layers
+      # actions/cache@v2.0.0
+      uses: actions/cache@b820478
+      with:
+        path: ${{ env.DOCKER_BUILDKIT_CACHE }}
+        key: ${{ runner.os }}-buildx-${{ matrix.target }}-${{ env.TAG }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-${{ matrix.target }}-
     - name: Build docker images
       env:
-        DOCKER_HOST: ssh://linkerd-docker
         DOCKER_TRACE: 1
       run: |
-        export PATH="`pwd`/bin:$PATH"
-        bin/docker-build
-  # todo: Keep in sync with `release.yml`
-  docker_push:
-    name: Docker push
-    runs-on: ubuntu-18.04
-    needs: [docker_build]
-    steps:
-    - name: Checkout code
-      # actions/checkout@v2
-      uses: actions/checkout@722adc6
-    - name: Set environment variables from scripts
-      run: |
-        . bin/_tag.sh
-        echo ::set-env name=TAG::$(CI_FORCE_CLEAN=1 bin/root-tag)
+        docker buildx create --driver docker-container --use
+        bin/docker-build-${{ matrix.target }}
     - name: Configure gcloud
```

Comment:
I don't think we want `Configure gcloud` as part of the matrix that this job is.

I think this will lead to each target of the matrix having a `Configure gcloud` step. Ideally, the `Push docker images to registry` remains part of this job, but then `Configure cloud` is a separate job that is dependent on `Docker build`.

Reply:
@kleimkuhler 

But how we can push the image if we don't run `Configure gcloud` beforehand?

Reply:
@aliariff Yep I was thinking this step was responsible for cluster creating as well which is not the case, so this should be fine!